sql/src/main/java/io/crate/operation/operator/any/AnyEqOperator.java

```diff
@@ -30,14 +30,17 @@
 import io.crate.sql.tree.ComparisonExpression;
 import io.crate.types.DataType;
 import io.crate.types.DataTypes;
-import io.crate.types.SetType;
+import io.crate.types.SingleColumnTableType;
 
 public class AnyEqOperator extends AnyOperator {
 
     public static final String NAME = OPERATOR_PREFIX + ComparisonExpression.Type.EQUAL.getValue();
 
     public static FunctionInfo createInfo(DataType targetType) {
-        return new FunctionInfo(new FunctionIdent(NAME, ImmutableList.of(targetType, new SetType(targetType))), DataTypes.BOOLEAN);
+        return new FunctionInfo(
+            new FunctionIdent(NAME,
+                ImmutableList.of(targetType, new SingleColumnTableType(targetType))),
```

Comment:
This was using `SetType` before which was also not correct. I think using the `ArrayType` should be ok. We can cast the `SingleColumnTableType` to `ArrayType`.